### DISCUSSION TITLE
RTSP session closing after pausing a gstreamer pipeline

### DISCUSSION THREAD

**Discussion by yoavos777** [bluenviron/mediamtx#2528] [Questions]
### Question

server version: MediaMTX v1.0.3
OS: Ubuntu 20.04.1
Gstreamer: 1.20.3

The problem:
Using gstreamer I am sending a stream to a rtsp address using the element rtspclientsink.
When I'm adding a new bin I set the pipeline to state PAUSE to link the new bin to the pipeline, This makes the, rtspclientsink element, to send a PAUSE message to MediaMTX.
Then immediately I set the pipeline back to state PLAYING, This makes the, rtspclientsink element, to send an ANNOUNCE message to MediaMTX.

Then MediaMTX prints the next log: 
INF [RTSP] [conn IP:PORT] closed(must be in state [initial], while is in state pre record).
I didnt fully understand What this log means.

**Comment by aler9** [comment]
Hello, that error message is raised because the transaction from PAUSE to ANNOUNCE is not supported by the server. Adding support for this feature is is too much of an effort since it's rarely used.

In order to solve the problem, delete and recreate rtspclientsink when you add a new bin.

**Comment by yoavos777** [comment]
Thank you for your response.

[a comment by an automated account is not shown]
